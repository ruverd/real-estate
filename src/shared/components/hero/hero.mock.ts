export const mockProps = {
  title: 'Test Title',
  description: 'Test Description',
  imageUrl: 'https://example.com/test-image.jpg',
};