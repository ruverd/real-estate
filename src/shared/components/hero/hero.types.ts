export interface HeroProps {
  title: string;
  description: string;
  imageUrl: string;
}