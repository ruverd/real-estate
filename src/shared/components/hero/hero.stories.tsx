import type { Meta, StoryObj } from "@storybook/react";
import { Hero } from "./hero";

const meta: Meta<typeof Hero> = {
  title: "Components/Hero",
  component: Hero,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {
    title: { control: "text" },
    description: { control: "text" },
    imageUrl: { control: "text" },
  },
};

export default meta;
type Story = StoryObj<typeof Hero>;

export const Default: Story = {
  args: {
    title: "Find Your Dream Home",
    description: "Discover the perfect property that fits your lifestyle and budget.",
    imageUrl: "https://images.unsplash.com/photo-1560518883-ce09059eeffa?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1773&q=80",
  },
};

export const WithShortTitle: Story = {
  args: {
    title: "Luxury Properties",
    description: "Exclusive high-end properties in prime locations.",
    imageUrl: "https://images.unsplash.com/photo-1600585154340-be6161a56a0c?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1770&q=80",
  },
};

export const WithLongDescription: Story = {
  args: {
    title: "Real Estate Investment",
    description: "Looking for investment opportunities? Our team of experts can help you find properties with high ROI potential. We analyze market trends, location value, and future development plans to ensure you make informed decisions.",
    imageUrl: "https://images.unsplash.com/photo-1582407947304-fd86f028f716?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1296&q=80",
  },
}; 