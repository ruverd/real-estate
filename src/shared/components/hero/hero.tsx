import { HeroProps } from "./hero.types";

export const Hero = ({ title, description, imageUrl }: HeroProps) => {
  return (
    <div 
      className="relative h-[400px] rounded-2xl overflow-hidden bg-cover bg-center bg-no-repeat"
      style={{
        backgroundImage: `url(${imageUrl})`,
      }}
      role="region"
      aria-label="hero section"
    >
      <div 
        className="absolute inset-0 bg-black/40" 
        aria-hidden="true" 
        data-testid="hero-overlay"
      />
      
      <div 
        className="relative z-10 h-full flex flex-col items-center justify-center gap-4 px-6"
        data-testid="hero-content"
      >
        <h1 className="text-4xl md:text-6xl text-center text-pretty font-sans font-bold text-white">{title}</h1>
        <p className="text-white text-pretty hidden md:block font-display text-center">{description}</p>
      </div>
    </div>
  );
};
