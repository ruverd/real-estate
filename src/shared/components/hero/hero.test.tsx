import { render, screen } from '@testing-library/react';

import { Hero } from './hero';
import { mockProps } from './hero.mock';
import type { HeroProps } from './hero.types';

const renderHero = (props: HeroProps) => {
  return render(<Hero {...props} />);
}

describe('Hero Component', () => {
  it('should renders the component with correct props', () => {
    renderHero(mockProps);
   
    expect(screen.getByText(mockProps.title)).toBeInTheDocument();
    expect(screen.getByText(mockProps.description)).toBeInTheDocument();
    
    const heroElement = screen.getByRole('region', { name: /hero/i });

    expect(heroElement).toHaveStyle(`background-image: url(${mockProps.imageUrl})`);
  }); 

  it('should renders without crashing when props are missing', () => {
    renderHero({ title: '', description: '', imageUrl: '' });

    expect(screen.getByRole('region', { name: /hero/i })).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderHero(mockProps);

    expect(container.firstChild).toMatchSnapshot();
  }); 
});
