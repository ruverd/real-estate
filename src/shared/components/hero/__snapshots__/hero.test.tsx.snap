// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero Component > should match snapshot 1`] = `
<div
  aria-label="hero section"
  class="relative h-[400px] rounded-2xl overflow-hidden bg-cover bg-center bg-no-repeat"
  role="region"
  style="background-image: url(https://example.com/test-image.jpg);"
>
  <div
    aria-hidden="true"
    class="absolute inset-0 bg-black/40"
    data-testid="hero-overlay"
  />
  <div
    class="relative z-10 h-full flex flex-col items-center justify-center gap-4 px-6"
    data-testid="hero-content"
  >
    <h1
      class="text-4xl md:text-6xl text-center text-pretty font-sans font-bold text-white"
    >
      Test Title
    </h1>
    <p
      class="text-white text-pretty hidden md:block font-display text-center"
    >
      Test Description
    </p>
  </div>
</div>
`;
